department-of-veterans-affairs/va.gov-cms
# VACMS 2452 Step By Step Component Alert Selection

## Description

See #2452 

## Testing done


## Screenshots


## QA steps

As a content editor
1. Login
2. Navigate here: `admin/structure/paragraphs_type`
 - [x] Confirm Alert Single, Non-reusable Alert paragraph types exist with structure outline below
        -New paragraph type "Non-reusable Alert" | machine name non_reusable_alert
            -fields (same fields that are presented for a non-reusable Alert in "Alert" paragraph type):
              -Alert Heading | field_alert_heading
              -Alert Content | field_va_paragraphs
               -Alert Type | field_alert_type
        -New paragraph type "Alert (single)" | machine name alert_single
          -fields
            - Alert Selection | field_alert_selection | three options "Reusable, "Non-reusable","None"
            - Reusable alert | field_alert_block_reference
            - Non-reusable Alert | field_alert_non_reusable_reference | reference to "Non-reusable Alert" paragraph
 - [x] Confirm updates to Step paragraph type:
      -"Alert" field_alert to "Step" paragraph, below WYSIWYG. Field can reference only paragraphs of type "Alert (single)"

3. Navigate to Alert section on form to confirm field behaviors `/node/add/step_by_step`
- [x] None is selected when creating new content by default
- [x] Reusable and Non-reusable fields are not visible until either Reusable or Non-reusable Alert Selections are selected.
- [x] If Reusable is selected and an entity reference is added, all alert selection radio buttons are disabled.
- [x] If Non-reusable is selected and a Non-reusable alert is added, all alert selection radio buttons are disabled.
- [x] When removing either the Reusable or Non-reusable alert, all the fields are then re-enabled.
- [x] Confirm same behaviors exist when adding new steps and alerts.
- [x] Confirm same behaviors exist when removing all steps or behaviors and re-adding a step and alert.
- [x] Save the node and confirm that the behaviors apply when returning to edit the node.

## Definition of Done
